### PR TITLE
chore: upgrade to Go 1.22

### PR DESCRIPTION
**Pull request by rootulp**
## Motivation

Part of https://github.com/celestiaorg/celestia-core/issues/1248
We want to cut a release of celestia-core with Go 1.22 support to enable celestia-node to bump to Go 1.22.

## Description

1. Upgrade to Go 1.22
2. Bump pyroscope deps
3. Copy over golangci-lint config from CometBFT v0.34.x branch
4. Resolve a few lint issues by copying code from CometBFT v0.34.x branch

### CODE REVIEW

**Comment by rootulp** [comment]
Still observing build errors:

```
Error: ../../../go/pkg/mod/github.com/pyroscope-io/godeltaprof@v0.1.2/internal/pprof/delta_mutex.go:25:20: undefined: runtime_cyclesPerSecond
Error: ../../../go/pkg/mod/github.com/pyroscope-io/godeltaprof@v0.1.2/internal/pprof/proto.go:400:8: undefined: runtime_expandFinalInlineFrame
```

perhaps I have to bump godeltaprof?

**Comment by rootulp** [comment]
I can reproduce the proto-check-breaking error locally:

```
$ make proto-check-breaking
Checking for breaking changes in Protobuf files against local branch
Note: This is only useful if your changes have not yet been committed.
      Otherwise read up on buf's "breaking" command usage:
      https://docs.buf.build/breaking/usage
# github.com/bufbuild/buf/private/pkg/observabilityzap
../../../../go/pkg/mod/github.com/bufbuild/buf@v1.15.1/private/pkg/observabilityzap/tracer_provider_closer.go:25:30: cannot use &tracerProviderCloser{} (value of type *tracerProviderCloser) as "go.opentelemetry.io/otel/trace".TracerProvider value in variable declaration: *tracerProviderCloser does not implement "go.opentelemetry.io/otel/trace".TracerProvider (missing method tracerProvider)
../../../../go/pkg/mod/github.com/bufbuild/buf@v1.15.1/private/pkg/observabilityzap/observabilityzap.go:33:25: cannot use tracerProvider (variable of type *tracerProviderCloser) as "go.opentelemetry.io/otel/trace".TracerProvider value in argument to otel.SetTracerProvider: *tracerProviderCloser does not implement "go.opentelemetry.io/otel/trace".TracerProvider (missing method tracerProvider)
```

It seems odd that v1.15.1 is used because the latest is much higher: 1.29.0. See https://github.com/bufbuild/buf/releases

**Comment by rootulp** [comment]
A bunch of linters are now failing but a few of them are disabled upstream. See https://github.com/cometbft/cometbft/blob/v0.34.x/.golangci.yml so we should disable revive and nakedret and likely fix the rest.

**Comment by rootulp** [comment]
`make lint` on the cometbft v0.34.x branch doesn't report any issues so I think we actually need to fix these in this repo.

**Review by evan-forbes — APPROVED** [review]
thanks!!

File: Makefile [hunk 1]
```diff
@@ -154,7 +154,7 @@ endif
 
 proto-gen: check-proto-deps
 	@echo "Generating Protobuf files"
-	@go run github.com/bufbuild/buf/cmd/buf generate
+	@go run github.com/bufbuild/buf/cmd/buf@v1.29.0 generate
```
Comment (staheri14) [inline Makefile:157]: [Suggestion] To facilitate future refactoring and considering that a specific version of this repo has been designated for use, I recommend introducing a variable to hold this version number and employing it throughout the Makefile in order to simplify version updates in the future.

File: .golangci.yml [hunk 2]
```diff
@@ -40,7 +51,74 @@ linters-settings:
     min-confidence: 0
   maligned:
     suggest-new: true
-
-run:
-  skip-files:
-    - libs/pubsub/query/query.peg.go
+  goconst:
+    ignore-tests: true
+  depguard:
```
Comment (staheri14) [inline .golangci.yml:56]: Nice one 👍

**Review by staheri14 — APPROVED** [review]
LGTM!

**Pull request merged**